proxy/src/main/java/com/wavefront/agent/logsharvesting/FlushProcessor.java

```diff
@@ -42,12 +85,9 @@ public void processTimer(MetricName name, Timer timer, FlushProcessorContext con
 
   @Override
   public void processGauge(MetricName name, Gauge<?> gauge, FlushProcessorContext context) throws Exception {
+    @SuppressWarnings("unchecked")
     ChangeableGauge<Double> changeableGauge = (ChangeableGauge<Double>) gauge;
```

Comment:
yes, this should only be used in conjunction with MetricCacheLoader. I made this class package private to be safer here